src/compas_timber/elements/fasteners/fastener.py

```diff
@@ -0,0 +1,182 @@
+from compas_model.elements import Element
+from compas_model.elements import reset_computed
+
+
+class Fastener(Element):
```

Comment:
I'm confused about this class. it re-defines some stuff from `Element` without implementing them. feels like whatever inherits from it should just inherit directly from `Element`.

Also why does a `Fastener` get features? I get it can have holes etc, but do we really need to represent them? 
In my opinion, we should be careful with the granularity level of what we represent and opt for very simple representation as a start.

Reply:
Another thought, should we maybe represent certain fasteners as just joints? for me that would fit because they would inflict drills and cuts to the beams they joint, kinda like the joints we already have. no?

Reply:
My thought is, they are applied as a type of joint. the joint would apply the processing aka Features as previous joint do, but the joint would also apply the `Fastener(Element)` objects as well.